src/components/ChromeLauncher.tsx

```
@@ -0,0 +1,340 @@
+  const swipeStartY = useRef(0);
+  const swiping = useRef(false);
+
+  const filteredApps = apps.filter((app) =>
+    app.name.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+
+  const totalPages = Math.max(1, Math.ceil(filteredApps.length / APPS_PER_PAGE));
+  const pageApps = useMemo(() => {
+    const start = currentPage * APPS_PER_PAGE;
+    return filteredApps.slice(start, start + APPS_PER_PAGE);
+  }, [filteredApps, currentPage]);
```

ESLint

[error] Compilation Skipped: Existing memoization could not be preserved

React Compiler has skipped optimizing this component because the existing manual memoization could not be preserved. The inferred dependencies did not match the manually specified dependencies, which could cause the value to change more or less frequently than expected. The inferred dependency was `APPS_PER_PAGE`, but the source dependencies were [filteredApps, currentPage]. Inferred different dependency than source.

/home/jailuser/git/src/components/ChromeLauncher.tsx:69:28
  67 |
  68 |   const totalPages = Math.max(1, Math.ceil(filteredApps.length / APPS_PER_PAGE));
> 69 |   const pageApps = useMemo(() => {
     |                            ^^^^^^^
> 70 |     const start = currentPage * APPS_PER_PAGE;
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 71 |     return filteredApps.slice(start, start + APPS_PER_PAGE);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 72 |   }, [filteredApps, currentPage]);
     | ^^^^ Could not preserve existing manual memoization
  73 |
  74 |   // Reset page when search changes or launcher opens
  75 |   useEffect(() => {

(react-hooks/preserve-manual-memoization)